### PR TITLE
[Data] Improve docstring and warning message for `from_huggingface`

### PR DESCRIPTION
**Pull request by amogkam**
Corrects return type hint, add docstring example, and log warning message for `from_huggingface`, per confusions from user feedback.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/read_api.py [hunk 1]
```diff
@@ -1797,12 +1824,22 @@ def convert(ds: "datasets.Dataset") -> Dataset:
         return ray_ds
 
     if isinstance(dataset, datasets.DatasetDict):
+        available_keys = list(dataset.keys())
+        logger.warning(
```
Comment (maxpumperla) [inline python/ray/data/read_api.py:1828]: this is great! I wonder if this explicit logic should also go into the docstring of `from_hugginface` directly (as a follow-up)

**Review by maxpumperla — APPROVED** [review]

**Review by bveeramani — APPROVED** [review]
LGTM

File: python/ray/data/read_api.py [hunk 2]
```diff
@@ -1772,20 +1772,47 @@ def from_spark(
...
+            schema={text: string, label: int64}
+        ), 'validation': MaterializedDataset(
+            num_blocks=1,
+            num_rows=374,
+            schema={text: string, label: int64}
+        )}
+        >>> ray_ds = ray.data.from_huggingface(hf_dataset["train"])
+        >>> ray_ds
+        MaterializedDataset(
+            num_blocks=1,
```
Comment (bveeramani) [inline python/ray/data/read_api.py:1804]: Is `num_blocks` gaurenteed to be 1? If not, you may want to use ellipsis to make the test determinsitic. https://docs.ray.io/en/master/ray-contribute/writing-code-snippets.html#ignoring-doctest-style-outputs

Comment (amogkam) [reply python/ray/data/read_api.py:1804]: yeah it is guaranteed to be 1 until we have distributed read support for `from_huggingface`

**Pull request merged**